ibis/backends/base/regex.py

```diff
@@ -0,0 +1,98 @@
+import re
+
+
+def normalize(r):
+    """Normalize a regular expression by ensuring that it is wrapped with:
+    '^' and '$'
+
+    Parameters
+    ----------
+    r : str or Pattern
+        The pattern to normalize.
+
+    Returns
+    -------
+    p : Pattern
+        The compiled regex.
+    """
+    r = getattr(r, "pattern", r)
+    return re.compile('^' + r.lstrip('^').rstrip('$') + '$')
+
+
+class RegexDispatcher:
```

Comment:
Nice utility! 

Could we place this to somewhere else, like `ibis.common.dispatch`?

Reply:
I'll move it there.  And yeah, it's slick!  @cpcloud pointed me to it.  It's from `odo`: https://github.com/blaze/odo/blob/master/odo/regex.py

Reply:
Then we need to reference it in the LICENSE file :)